sm/tedge_agent/tests/main.rs

```diff
@@ -0,0 +1,39 @@
+#[cfg(test)]
+#[cfg(feature = "mosquitto-available")]
+mod tests {
+    use std::process::{Command, Stdio};
+    use std::{thread, time};
+
+    use assert_cmd::prelude::*;
+    use predicates::prelude::*;
+
+    #[test]
+    /// Tests that only one instance of `tedge_agent` is running.
+    /// This is done by spawning/running two instances of `tedge_agent`
+    /// expecting the first one to work and the second to fail.
+    fn tedge_agent_check_no_multiple_instances_running() -> Result<(), Box<dyn std::error::Error>> {
+        // running first `tedge_agent` binary
+        let mut agent = Command::cargo_bin(env!("CARGO_PKG_NAME"))?
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()?;
+
+        // NOTE: if this sleep is removed, there seems to be a race-condition
+        // between `agent` and `agent_2`. A 10 milisecond delay prevents this.
+        let ten_millis = time::Duration::from_millis(10);
+        thread::sleep(ten_millis);
```

Comment:
This is intriguing and deserves more in-depth understanding since the point of this lock is to avoid races!

Reply:
I doubled check. The race condition is normal and the comment needs to be changed.

```
// A sleep is required here to be sure that the lock is acquired by the first agent and not the second.
```

Reply:
An other option would be to remove the race from the tests by checking that one of the two agents has been stopped by the lock.